app/client/src/reducers/uiReducers/applicationsReducer.tsx

```diff
@@ -214,6 +216,30 @@ const applicationsReducer = createReducer(initialState, {
       forkingApplication: false,
     };
   },
+  [ReduxActionTypes.IMPORT_APPLICATION_INIT]: (
+    state: ApplicationsReduxState,
+  ) => {
+    return { ...state, importingApplication: true };
+  },
```

Comment:
Nitpick: can use shorthand here
```suggestion
  ) => ({ ...state, importingApplication: true }),
```